### ISSUE TITLE
[BUG] Stuck in high resistance and Zwift Ride controllers don't do anything in Rouvy

### ISSUE THREAD

**Issue by gengelke** [cagnulein/qdomyos-zwift#2666]
**Describe the bug**
Wattage is immediately increased until I can not pedal any more. Zwift Ride controllers don't do anything.

**To Reproduce**
1. I have started the QZ app on my iPhone, opened the settings and enabled Zwift Click and Zwift Play options in Accessories->Zwift Devices Options
2. The app then tells me that Zwift Ride/Play has been connected and both blue Z buttons of the Zwift Ride controllers constantly glow bue. So everything looks as expected
3. Then I started the Rouvy app on my iPad and connected the smart trainer which is named like my iPhone.
4. I picked a route in Rouvy and started it. When I start to pedal the cadence and wattage are shown in the display of Rouvy. Unfortunately the resistance then is increased until I can not pedal any more. This appears always after some seconds. When I stop pedaling it recovers but when I continue to pedal it gets stuck again in way too high wattage. Furthermore neither my Zwift Ride controllers nor my Zwift Click didn't do anything when I press their buttons.

**Expected behavior**
Resistance based on gradient and speed ;-)
Working Zwift Ride controller virtual shifting 

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone 13 mini
 - OS: iOS
 - Version 17.6.1
I did the same on an iPad but same behavior. Rouvy was running either on iPad or Apple TV. Same behavior.

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
[debug-Mi__Okt__16_09_33_55_2024.log](https://github.com/user-attachments/files/17391531/debug-Mi__Okt__16_09_33_55_2024.log)

  
**Additional context**

**Comment by cagnulein** [comment]
@gengelke in the log i saw that you are keep pushing the gear+ on the zwift ride, this is forcing the resistance to the star. why are you doing so? just to be clear: gear positive means more resistance, gears negative means less resistance. Let me know

**Comment by gengelke** [comment]
https://youtube.com/shorts/oLCZxsPaXro?feature=share

**Comment by gengelke** [comment]
I just repeated the steps but this time I have disabled Zwift Ride/Play stuff in QZ but left Zwift Click enabled. In this scenario virtual shifting with Zwift Click works as expected and therefore the resistance is increased/decreased accordingly when I press +/- on the Zwift Click buttons. So obviously there's something wrong with the Zwift Ride controller connection.

**Comment by cagnulein** [comment]
hah so you have both zwift play controller and zwift click? which one do you want to use with QZ? I guess you are the first one with this scenario! Let me know

**Comment by cagnulein** [comment]
also are the zwift play updated to the latest firmware?

**Comment by cagnulein** [comment]
> Isn't this supported with QZ?

it should but I never tried :)

ok let me check again the log, it seems that your zwift play are sending a different messages from all the other one that i already handled. it seems like the button are pressing several times...

**Comment by gengelke** [comment]
It's hardware version B.0 if this helps

**Comment by cagnulein** [comment]
@gengelke one important note @gengelke qz handled both brake buttons and gear buttons of the zwift ride as gear buttons for qz.

What I'm seeing in the log it seems that your brake button is stucked, is it possible? actually not stuck, it just micro events continuosly. also it seems that after 60 seconds, it does the same but on the other controller.

**Comment by cagnulein** [comment]
can you post a debug log where you connect the zwift ride only but you don't touch nothing at all? in the comment above you said so but you linked the same debug log of the original post

thanks

**Comment by gengelke** [comment]
Oh, my mistake. Should be this one: 
[Uploading debug-Mi__Okt__16_10_12_02_2024.log…]()

**Comment by cagnulein** [comment]
you have to wait the finish of the upload before commenting otherwise the log is corrupted. thanks

**Comment by gengelke** [comment]
Next try... ;-) I recognized that when I don't touch anything after I have started a new ride in Rouvy I can pedal as expected. Did this for about 30sec. Then I had to shift because the resistance was to low. Therefore I pressed the shifting buttons of the Zwift Ride controller on the right side. Which then led to the same behavior as described before: resistance is stuck

[latest_debug-Mi__Okt__16_12_04_59_2024.log](https://github.com/user-attachments/files/17393773/latest_debug-Mi__Okt__16_12_04_59_2024.log)

**Comment by cagnulein** [comment]
could you please enable the gear debouncing in the zwift accesories setting in QZ? because now i'm seeing everything correctly, you have only to enable this one to avoid the bouncing of the buttons

let me know!

**Comment by cagnulein** [comment]
note for me: it's the 0xfb code that it seems stuck on this user, in case I can try to filter it out with a special setting

**Comment by gengelke** [comment]
Same again, I can pedal at the beginning. Then I press the shifter button and it gets stuck.
[debug-Mi__Okt__16_12_30_11_2024.log](https://github.com/user-attachments/files/17394449/debug-Mi__Okt__16_12_30_11_2024.log)

**Comment by cagnulein** [comment]
That's so strange, are you 100% sure that the zwift ride buttons are
working always correctly on zwift directly?

I will try to examine it better ASAP if I could do something : because in
this log it seems the same exact frame sent from the play when you click
it, it's the same that it's sending over and over

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 12:49 Gordon Engelke <
***@***.***> ha scritto:

**Comment by gengelke** [comment]
Well, I own Zwift Ride since about three weeks and have done 400km with it. Didn't notice any failures yet. Each session was around 2-2.5h so I assume that I would have noticed if the buttons wouldn't work in between. But take your time. Maybe you'll have an idea later and/or someone else has similar issues and maybe provides additional data. I normally use Zwift so it's currently no party crasher if can't use Rouvy or one of the others ;-) Just wanted to give it a try because QZ seems pretty promising for poeple who own/want a Zwift Ride but don't want to be locked-in to Zwift :-)

**Comment by cagnulein** [comment]
Ok I will check the log and I will be back to you!

**Comment by cagnulein** [comment]
@gengelke i was trying to solve your mistery.
On the last log i saw this gears changing

```
Mi. Okt. 16 12:31:12 2024 1729074672818  setGears 1
Mi. Okt. 16 12:31:18 2024 1729074678282  setGears 2
Mi. Okt. 16 12:31:22 2024 1729074682542  setGears 3
Mi. Okt. 16 12:31:25 2024 1729074685121  setGears 4
Mi. Okt. 16 12:31:28 2024 1729074688179  setGears 5
Mi. Okt. 16 12:31:31 2024 1729074691060  setGears 6
Mi. Okt. 16 12:31:31 2024 1729074691602  setGears 7
Mi. Okt. 16 12:31:31 2024 1729074691897  setGears 8
Mi. Okt. 16 12:31:33 2024 1729074693762  setGears 9
Mi. Okt. 16 12:31:38 2024 1729074698797  setGears 10
Mi. Okt. 16 12:31:39 2024 1729074699406  setGears 11
Mi. Okt. 16 12:31:39 2024 1729074699705  setGears 12
Mi. Okt. 16 12:31:39 2024 1729074699942  setGears 13
Mi. Okt. 16 12:31:40 2024 1729074700182  setGears 14
Mi. Okt. 16 12:31:42 2024 1729074702639  setGears 15
Mi. Okt. 16 12:31:43 2024 1729074703061  setGears 16
Mi. Okt. 16 12:31:43 2024 1729074703418  setGears 17
Mi. Okt. 16 12:31:43 2024 1729074703485  setGears 18
Mi. Okt. 16 12:31:43 2024 1729074703783  setGears 19
Mi. Okt. 16 12:31:44 2024 1729074704441  setGears 20
Mi. Okt. 16 12:31:45 2024 1729074705278  setGears 19
Mi. Okt. 16 12:31:45 2024 1729074705645  setGears 18
Mi. Okt. 16 12:31:46 2024 1729074706298  setGears 17
Mi. Okt. 16 12:31:46 2024 1729074706718  setGears 16
Mi. Okt. 16 12:31:47 2024 1729074707198  setGears 15
Mi. Okt. 16 12:31:50 2024 1729074710683  setGears 16
Mi. Okt. 16 12:31:51 2024 1729074711397  setGears 17
Mi. Okt. 16 12:31:52 2024 1729074712278  setGears 18
```

which one do you recognize that you pressed yourself? because I compared your log with a random one from another QZ zwift ride user and when the zwift ride is reporting the button pressed on your log is sending the same frame that it sends from the other user. So it doesn't seems to me a QZ issue, but I really want to solve this mistery!

Could you please record a short video with the issue showing me gear tile on qz and your zwift ride togheter along with a debug log? So I can try to decypher which one are the fake touch and which ones are not. You can also enable the timestamp tile to show me also the timestamp on QZ

let me know

**Comment by gengelke** [comment]
@cagnulein Thanks for the info. I'll try to recreate the scenario on the weekend when I've more time and will keep you informed and provide additional data

**Comment by cagnulein** [comment]
Sounds good, thanks!


Il giorno ven 18 ott 2024 alle 00:17 Gordon Engelke <
***@***.***> ha scritto:

**Comment by gengelke** [comment]
First of all I just quickly put the QZ app to my older iPhone X to see if it makes a difference. Virtual shifting with my Zwift Ride controllers seems to work in this scenario. So the issues described above seem to be related with my current iPhone 13 mini in combination with QZ

[debug-Sa__Okt__19_08_42_39_2024.log](https://github.com/user-attachments/files/17443821/debug-Sa__Okt__19_08_42_39_2024.log)

**Comment by gengelke** [comment]
I've now used my iPhone 13 mini again but I am not able to reproduce the issue. It now behaves like with iPhone X. When pressing the shifting buttons I see the gear tile to increase/decrease accordingly. Both with Zwift Ride controllers as well as Zwift Click. The only thing I recognized with both iPhone scenarios is that even when the tile displays the gear change, it still feels like the same resistance for quite some time. So it felt not as quick as I would expect because I was already on top of the mountain when gears adapted. But maybe I just have to get used to it a while. I could not make a video because I did not know how to shift down with the left hand, shift up with the right hand and hold the camera at the same time ;-) But hopefully the log is helpful anyway.

[debug-Sa__Okt__19_10_21_03_2024.log](https://github.com/user-attachments/files/17444435/debug-Sa__Okt__19_10_21_03_2024.log)
